main.go

```diff
@@ -18,11 +20,196 @@ import (
 	"github.com/nats-io/nats.go"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/viper"
+	"go.opentelemetry.io/otel"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
 var log = logging.Log.WithFields(logrus.Fields{"service": "discoenv-users-service"})
 
+// NATSMessageHandler is the handler type that we support. Should map to the nats.Handler
+// function signature func(*nats.Msg).
+type NATSMessageHandler func(context.Context, *nats.Msg)
+
+type NATSTextMapCarrier struct {
+	nats.Header
+}
```

Comment:
I'm not totally sure why you're not just doing `type NATSTextMapCarrier nats.Header`. Maybe doesn't make a difference? not sure

Reply:
This code is handled in a different repository now.